curriculum/challenges/english/03-front-end-libraries/jquery/target-html-elements-with-selectors-using-jquery.english.md

```diff
@@ -32,8 +32,7 @@ tests:
   - text: You should only use jQuery to add these classes to the element.
     testString: 'assert(!code.match(/class.*animated/g));'
   - text: Your jQuery code should be within the <code>$(document).ready();</code> function.
-    testString: assert(code.match(/\$\(document\)\.ready\(function.*(\s|\n)*.*button.*.addClass.*\);/g));
-
+    testString: assert(code.match(/\$\(document\)\.ready\(function\s*\(\)\s*{\n\s*\$.*button.*.addClass.*/g));
```

Comment:
```suggestion
```

Reply:
Not sure how it is redundant.  Based on the instructions, we expect the user to put the other code inside the `$(document).ready() function`.  The first test only checks if they are using the the `addClass` method regardless of where it is placed.

Reply:
The first test is:
```
assert($("button").hasClass("animated") && $("button").hasClass("bounce"));
```
Does this not use jQuery to test if the class has been successfully added to the `button` element?

Reply:
@Sky020 Yes, but it does not validate that the `hasClass` method code is inside the `$(document).ready()` function, which is part of the instructions.

Reply:
The user is expected to use the `addClass` method to add two classes to the `button` element. So, if the `button` has those two classes added to it, either the user has used DOM methods to add the class (which requires adding a script tag to the bottom of the editor with the code, and they would need to do something similar to:
```
<script>
  document.querySelectorAll('button').forEach(ele => {
    ele.classList
    .add('animated');
    // The random spacing is necessary to fool the second test
    ele.classList
    .add('bounce');
  });
</script>
```
, **or** they have placed the `addClass` code within the `document.ready` function. If the `addClass` method is not within the `document.ready` function, the first test will not pass. Unless code like this has been written:
```
  $(document).ready(function() {
    callback()
  });
  function callback() {
    $('button').addClass("animated bounce");
  }
```

I think fooling this test would be an acceptable way to pass. That is why I think it is redundant.